superset-frontend/plugins/plugin-chart-echarts/src/BoxPlot/buildQuery.ts

```diff
@@ -16,26 +16,40 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { buildQueryContext } from '@superset-ui/core';
+import {
+  AdhocColumn,
+  buildQueryContext,
+  ensureIsArray,
+  isPhysicalColumn,
+} from '@superset-ui/core';
 import { boxplotOperator } from '@superset-ui/chart-controls';
 import { BoxPlotQueryFormData } from './types';
 
 export default function buildQuery(formData: BoxPlotQueryFormData) {
-  const { columns = [], granularity_sqla, groupby = [] } = formData;
-  return buildQueryContext(formData, baseQueryObject => {
-    const distributionColumns: string[] = [];
-    // For now default to using the temporal column as distribution column.
-    // In the future this control should be made mandatory.
-    if (!columns.length && granularity_sqla) {
-      distributionColumns.push(granularity_sqla);
-    }
-    return [
-      {
-        ...baseQueryObject,
-        columns: [...distributionColumns, ...columns, ...groupby],
-        series_columns: groupby,
-        post_processing: [boxplotOperator(formData, baseQueryObject)],
-      },
-    ];
-  });
+  return buildQueryContext(formData, baseQueryObject => [
+    {
+      ...baseQueryObject,
+      columns: [
+        ...ensureIsArray(formData.columns).map(col => {
+          if (
+            isPhysicalColumn(col) &&
+            formData.time_grain_sqla &&
+            formData?.datetime_columns_lookup?.[col]
+          ) {
+            return {
+              timeGrain: formData.time_grain_sqla,
+              columnType: 'BASE_AXIS',
+              sqlExpression: col,
+              label: col,
+              expressionType: 'SQL',
+            } as AdhocColumn;
+          }
```

Comment:
Do we want to check for the `GENERIC_CHART_AXES` FF here similar to what we're doing in the other plugins?

Reply:
Boxplot is a bit special. the `Distribute Across` is similar to `XAxis` control in time series charts, but the `Distribute Across` control is shown even though FF is disabled(the original logic). So the build query logic skips checking FF, which means always transforming the Physical Column into BaseAxisAdhocColumn.

Additionally, the `Distribute Across` control use the `initialValue` property to set the initial value as the `default time column`, so the BoxPlot is "native" to support generic xaxis.

Reply:
Got it, thanks for the explanation. There may be the special case where a user wants to apply a time grain on "Dimensions" but not "Distribute across" or vice versa. But maybe that can be addressed once time grain is moved into the column control.